src/Nethermind.Int256/UInt256.cs

```diff
@@ -704,36 +721,36 @@ private static void Udivrem(ref ulong quot, ref ulong u, int length, in UInt256
             un[uLen] = Rsh(Unsafe.Add(ref u, uLen - 1), 64 - shift);
             for (int i = uLen - 1; i > 0; i--)
             {
-                un[i] = Lsh(Unsafe.Add(ref u, i), shift) | Rsh(Unsafe.Add(ref u, i - 1), 64 - shift);
+                un[i] = NativeLsh(Unsafe.Add(ref u, i), shift) | Rsh(Unsafe.Add(ref u, i - 1), 64 - shift);
             }
 
-            un[0] = Lsh(u, shift);
+            un[0] = NativeLsh(u, shift);
 
             // TODO: Skip the highest word of numerator if not significant.
 
             if (dLen == 1)
             {
-                ulong dnn0 = Lsh(d.u0, shift);
+                ulong dnn0 = NativeLsh(d.u0, shift);
                 ulong r = UdivremBy1(ref quot, un, dnn0);
-                r = Rsh(r, shift);
+                r = NativeRsh(r, shift);
                 rem = (UInt256)r;
                 return;
             }
 
-            ulong dn0 = Lsh(d.u0, shift);
+            ulong dn0 = NativeLsh(d.u0, shift);
             ulong dn1 = 0;
             ulong dn2 = 0;
             ulong dn3 = 0;
             switch (dLen)
             {
                 case 4:
-                    dn3 = Lsh(d.u3, shift) | Rsh(d.u2, 64 - shift);
+                    dn3 = NativeLsh(d.u3, shift) | Rsh(d.u2, 64 - shift);
```

Comment:
I might be not getting it, but can't we `Rsh` natively here as well?

Reply:
I think we can't guarantee in this case that `shift != 0`

Reply:
If so - then `64 - shift` would be an issue. That's why it's like that

Reply:
@Scooletz 
Hey! Just double-checking - did it make sense now? I would also like to change those if possible, but can't see it if so.